smart-contracts/cosmwasm-contract/src/contract.rs

```diff
@@ -0,0 +1,231 @@
+    });
+    let verify_oracle_proof = cosmwasm_std::WasmMsg::Execute {
+        contract_addr: query_supra_pull_contract(deps.as_ref())
+            .unwrap()
+            .supra_pull_contract,
+        msg: Binary::from(serde_json::to_vec(&execute_msg).unwrap()),
+        funds: vec![],
+    };
+
+    // Creating a submessage that wraps the message above
+    let submessage = SubMsg::reply_on_success(verify_oracle_proof, VERIFY_PROOF_REPLY_ID);
+    // Creating a response with the submessage
+    let response = Response::new().add_submessage(submessage);
+    Ok(response)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::GetSupraPullContract {} => to_json_binary(&query_supra_pull_contract(deps)?),
+        QueryMsg::GetPairDataInternal { pair_id } => {
+            to_json_binary(&query_pair_data_internal(deps, pair_id)?)
+        }
+        QueryMsg::GetPairDataSupra { pair_id } => {
+            to_json_binary(&query_pair_data_supra(deps, pair_id)?)
+        }
+    }
+}
+
+fn query_supra_pull_contract(deps: Deps) -> StdResult<CountResponse> {
+    let state = STATE.load(deps.storage)?;
+    Ok(CountResponse {
+        supra_pull_contract: state.supra_pull_contract,
+    })
+}
+fn query_pair_data_supra(deps: Deps, pair_id: u32) -> StdResult<OracleHolder> {
+    let query_message = json!({
+        "get_svalue" : {
+            "pair_index" : pair_id
+        }
+    });
+    let state = STATE.load(deps.storage)?;
+
+    let response: StdResult<OracleHolder> = deps
+        .querier
+        .query_wasm_smart(state.supra_pull_contract, &query_message);
+    response
+}
+
+fn query_pair_data_internal(deps: Deps, pair_id: u32) -> StdResult<OracleHolder> {
+    let state = STATE.load(deps.storage)?;
+    if let Some(oracle_holder) = state.price_data.get(&pair_id) {
+        Ok(oracle_holder.clone())
+    } else {
+        panic!("Pair id not found");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cosmwasm_std::testing::{mock_dependencies_with_balance, mock_env, mock_info};
+    use cosmwasm_std::{coins, from_binary};
+
+    #[test]
+    fn proper_initialization() {
+        let mut deps = mock_dependencies_with_balance(&coins(2, "token"));
+
+        let msg = InstantiateMsg {
+            supra_pull_contract: "Test".to_string(),
+        };
+        let info = mock_info("creator", &coins(1000, "earth"));
+
+        // we can just call .unwrap() to assert this was a success
+        let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+        assert_eq!(0, res.messages.len());
+
+        // it worked, let's query the state
+        let res = query(deps.as_ref(), mock_env(), QueryMsg::GetSupraPullContract {}).unwrap();
+        let value: CountResponse = from_binary(&res).unwrap();
+        assert_eq!("Test".to_string(), value.supra_pull_contract);
+    }
+
+    #[test]
+    fn increment() {
```

Comment:
The function name does not accurately reflect the tests performed within it. Please update the function name to describe better the tests being conducted.

Reply:
And also you can move all the test cases into the `integration_tests.rs` file